### PR TITLE
fix wrong comments of dra

### PR DESCRIPTION
**Pull request by Monokaix**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Some dra feature related code comments are confusing, maybe cause by code copy, just correct them. 
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by pohly — APPROVED** [review]
/lgtm
/approve

Thanks!

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -3929,7 +3929,7 @@ type PodResourceClaimStatus struct {
 	Name string `json:"name" protobuf:"bytes,1,name=name"`
 
 	// ResourceClaimName is the name of the ResourceClaim that was
-	// generated for the Pod in the namespace of the Pod. It this is
+	// generated for the Pod in the namespace of the Pod. If this is
```
Comment (bart0sh) [inline staging/src/k8s.io/api/core/v1/types.go:3932]: This should be changed in the .proto file and .go files should be re-generated.

Comment (pohly) [reply staging/src/k8s.io/api/core/v1/types.go:3932]: No, for the Kubernetes API, the `types.go` files are the source. One has to run various update scripts to propagate this fix to other generated files.

@Monokaix: this is not worth the time that we would have to spend on getting API approval. Can you take out this part? I'll add it to a [pending API update PR](https://github.com/kubernetes/kubernetes/pull/124075).

File: pkg/controller/resourceclaim/controller.go [hunk 2]
```diff
@@ -547,7 +547,7 @@ func (ec *Controller) syncPod(ctx context.Context, namespace, name string) error
 	return nil
 }
 
-// handleResourceClaim is invoked for each volume of a pod.
+// handleResourceClaim is invoked for each resource claim of a pod.
```
[the remark this answers is not in the document]
Comment (Monokaix) [reply pkg/controller/resourceclaim/controller.go:550]: Checked that, no other errors found: ）

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (Monokaix) [reply staging/src/k8s.io/api/core/v1/types.go]: It's good idea, have removed that: ) @pohly

File: pkg/controller/resourceclaim/controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bart0sh) [reply pkg/controller/resourceclaim/controller.go:550]: Thanks.
/lgtm

**Comment by Monokaix** [comment]
Hi, ci shows that pr must be in milestone v1.30, can this be solved?

**Pull request merged**